### DISCUSSION TITLE
Can't get current user if set an alias on localhost

### DISCUSSION THREAD

**Discussion by Tur8008** [fastapi-users/fastapi-users#1418] [Q&A]
Hello.
I experienced with quite a weird behavior once had added an alias on localhost in my /etc/hosts (I need to to that for testing purposes cause cypress doesn't work properly in case when frontend and backend use the same domain and different ports)
like
```shell
127.0.0.1    localhost    back.dev.local
```
Since then once the app gets `HTTP 204` on `http://back.dev.local:8000/auth/jwt/login`
<img width="673" alt="Снимок экрана 2024-07-16 в 11 44 29" src="https://github.com/user-attachments/assets/a0e4be39-c4be-4d6a-907e-80f016428eac">

I get 401 -- Unauthorized on request to `http://back.dev.local:8000/users/me`
<img width="442" alt="Снимок экрана 2024-07-16 в 11 48 27" src="https://github.com/user-attachments/assets/57c57912-7b85-4b8c-bdf4-c75514f1931f">

It seems like the problem is in 
```python
get_current_active_user = authenticator.current_user(active=True, verified=requires_verification)
```
because `/users/me` endpoint  returns `HTTP 401`
```python
 @router.get(
        "/me",
        response_model=user_schema,
        name="users:current_user",
        responses={
            status.HTTP_401_UNAUTHORIZED: {
                "description": "Missing token or inactive user.",
            },
        },
    )
    async def me(
        user: models.UP = Depends(get_current_active_user),
    ):
        result = user_schema.from_orm(user)
        logger.debug("Got user: %s", result)
        return result
        ```
   I was wandering  if you can get any idea cause if I return back to localhost:8000 it starts working fine

**Comment by frankie567** [comment]
It probably has to do with the cookie configuration: https://fastapi-users.github.io/fastapi-users/latest/configuration/authentication/transports/cookie/#configuration

Intuitively, I would say that if you let `cookie_secure` flag to `True`, then it won't work because `http://back.dev.local:8000/` is not served under HTTPS. The reason it works with `localhost` is because Chrome is nice enough to relax that constraint for `localhost` (which is not the case of every browsers).

**Reply by Tur8008** [reply]
So I just need to set up cookie secure false just as dev setup, do I?

**Reply by frankie567** [reply]
Yes! I recommend you to put this as a setting (like an environment variable) to easily change it.

In production, it's **critical** to keep it to `True`.

**Reply by Tur8008** [reply]
I've already had it instaled
<img width="1036" alt="Снимок экрана 2024-07-16 в 12 42 51" src="https://github.com/user-attachments/assets/488fd4a7-8b48-4fd6-ae60-78d0ae824876">
And` DEBUG` is true

**Reply by Tur8008** [reply]
May be I need to change `cookie_samesite` param

**Reply by frankie567** [reply]
You should inspect the **Cookies** tab of the Chrome network inspector, both for the login request and then the authenticated request. 

You should see the characteristics of the set cookie, and hence understand why it's not passed in the authenticated request.

**Comment by Tur8008** [comment]
Yes, you are absolutely right! I've just found out that get request on /users/me did not send the cookie
